nixos/modules/services/networking/wireguard.nix

```diff
@@ -255,38 +270,43 @@ let
           ${values.preSetup}
 
           ip link add dev ${name} type wireguard
+          ${lib.optionalString (!isNull values.namespace) ''
+            ip netns add "${values.namespace}"
+            ip link set ${name} netns "${values.namespace}"
+          ''}
 
           ${concatMapStringsSep "\n" (ip:
-            "ip address add ${ip} dev ${name}"
+            "${ipns} address add ${ip} dev ${name}"
           ) values.ips}
 
-          wg set ${name} private-key ${privKey} ${
+          ${wgns} set ${name} private-key ${privKey} ${
             optionalString (values.listenPort != null) " listen-port ${toString values.listenPort}"}
 
           ${concatMapStringsSep "\n" (peer:
             assert (peer.presharedKeyFile == null) || (peer.presharedKey == null); # at most one of the two must be set
             let psk = if peer.presharedKey != null then pkgs.writeText "wg-psk" peer.presharedKey else peer.presharedKeyFile;
             in
-              "wg set ${name} peer ${peer.publicKey}" +
+              "${wgns} set ${name} peer ${peer.publicKey}" +
               optionalString (psk != null) " preshared-key ${psk}" +
               optionalString (peer.endpoint != null) " endpoint ${peer.endpoint}" +
               optionalString (peer.persistentKeepalive != null) " persistent-keepalive ${toString peer.persistentKeepalive}" +
               optionalString (peer.allowedIPs != []) " allowed-ips ${concatStringsSep "," peer.allowedIPs}"
             ) values.peers}
 
-          ip link set up dev ${name}
+          ${ipns} link set up dev ${name}
 
           ${optionalString (values.allowedIPsAsRoutes != false) (concatStringsSep "\n" (concatMap (peer:
               (map (allowedIP:
-                "ip route replace ${allowedIP} dev ${name} table ${values.table}"
+                "${ipns} route replace ${allowedIP} dev ${name} table ${values.table}"
```

Comment:
The [wireguard netns guide](https://www.wireguard.com/netns/) does 

```
ip -n container route add default dev wg0
```

Is this a relevant difference?

Reply:
if allowedIP is 0.0.0.0/0, then no. Otherwise, ours will only add relevant routes, while the wireguard example always routes all traffic. People wanting to route all traffic generally have `allowedIPs set to ["0.0.0.0/0", "::0/0"]`.